### PR TITLE
[performance] remove last of relational queries to instead rely on caches

### PR DESCRIPTION
**Pull request by NyaaaWhatsUpDoc**
# Description

- Removes relational part of queries for status selects and user selects
- Adds new separate `applicationDB{}` with its own caching

closes https://github.com/superseriousbusiness/gotosocial/issues/1094

## Checklist
- [x] I/we have read the [GoToSocial contribution guidelines](https://github.com/superseriousbusiness/gotosocial/blob/main/CONTRIBUTING.md).
- [x] I/we have discussed the proposed changes already, either in an issue on the repository, or in the Matrix chat.
- [x] I/we have performed a self-review of added code.
- [x] I/we have written code that is legible and maintainable by others.
- [x] I/we have commented the added code, particularly in hard-to-understand areas.
- [x] I/we have made any necessary changes to documentation.
- [x] I/we have added tests that cover new code.
- [x] I/we have run tests and they pass locally with the changes.
- [x] I/we have run `go fmt ./...` and `golangci-lint run`.

### CODE REVIEW

File: internal/middleware/tokencheck.go [hunk 1]
```diff
@@ -125,15 +124,16 @@ func TokenCheck(dbConn db.DB, validateBearerToken func(r *http.Request) (oauth2.
 			log.Tracef(ctx, "authenticated client %s with bearer token, scope is %s", clientID, ti.GetScope())
 
 			// fetch app for this token
-			app := &gtsmodel.Application{}
-			if err := dbConn.GetWhere(ctx, []db.Where{{Key: "client_id", Value: clientID}}, app); err != nil {
+			app, err := dbConn.GetApplicationByClientID(ctx, clientID)
```
Comment (tsmethurst) [inline internal/middleware/tokencheck.go:127]: Much better, my goodness those old `GetWhere` things are clunky looking.

**Review by tsmethurst — APPROVED** [review]
Looks good to me! Feel free to squerge! :partying_face:

**Comment by NyaaaWhatsUpDoc** [comment]
> Looks good to me! Feel free to squerge! partying_face

damn that was quick :O

**Comment by tsmethurst** [comment]
> HOW IS THIS NOT PASSING I LITERALLY RAN THIS LOCALLY

stinky...

**Comment by NyaaaWhatsUpDoc** [comment]
> > HOW IS THIS NOT PASSING I LITERALLY RAN THIS LOCALLY
> 
> stinky...

downvote

**Comment by NyaaaWhatsUpDoc** [comment]
Right, so the issue this has revealed, is that when we delete statuses they are then not able to be included in the report after performing the suspension. (which i think is a similar case on Mastodon?). Previously we were saved by the test being just fast enough to beat-out generating the report before the enqueued status deletes, but now the test is failing the expected JSON.

So the solution here is either we update the test's expected JSON output, or we ignore deleting statuses if they're in a report.

**Pull request merged**